Source/Modules/fortran.cxx

```diff
@@ -0,0 +1,4101 @@
+    generic = true;
+  } else if (String *manual_name = Getattr(n, "fortran:name")) {
+    // Override the fsymname name for this function
+    assert(!fsymname);
+    fsymname = Copy(manual_name);
+  }
+
+  // Add suffix if the function is overloaded (can't overload C bound functions)
+  if (String *overload_ext = (Getattr(n, "sym:overloaded") ? Getattr(n, "sym:overname") : NULL)) {
+    Append(wname, overload_ext);
+    Append(imname, overload_ext);
+    if (!fsymname) {
+      // Overloaded functions become fsymname
+      fsymname = fname;
+      fname = proxy_name_construct(this->getNSpace(), symname);
+    }
+    fname = ensure_short(fname, 63 - Len(overload_ext));
+    Append(fname, overload_ext);
+    generic = true;
+  }
+
+  // Check for conflicts with other member functions in parent classes:
+  // - If the lowercase/renamed function name is the same as any other class
+  //   function (or base class's function), print a warning and ignore it.
+  // - If the function shadows another function, the signature must match
+  //   exactly: if the parent is 'generic', this one must be as well. If it's a
+  //   subroutine, same deal.
+  if (member) {
+    assert(fsymname);
+    String *lower_fsymname = Swig_string_lower(fsymname);
+    Symtab *fsymtab = Getattr(this->getCurrentClass(), "fortran:symtab");
+    Node *overload_base = NULL;
+
+    // Check against lowercase name conflicts
+    for (Symtab *st = fsymtab; st; st = parentNode(st)) {
+      if (Node *other = Getattr(st, lower_fsymname)) {
+        String *other_fsymname = Getattr(other, "wrap:fsymname");
+        if (!other_fsymname) {
+          other_fsymname = Getattr(other, "fortran:name");
+        }
+        if (other_fsymname && !Equal(other_fsymname, fsymname)) {
+          // It's actually a different identifier
+          Swig_warning(WARN_FORTRAN_NAME_COLLISION, input_file, line_number,
+                       "Ignoring '%s' due to name conflict with '%s'\n",
+                       fsymname, other_fsymname);
+          Swig_warning(WARN_FORTRAN_NAME_COLLISION, Getfile(other), Getline(other),
+                       "Previous declaration of '%s'\n",
+                       Getattr(other, "sym:name"));
+          SetFlag(n, "fortran:ignore");
+          return SWIG_NOWRAP;
+        } else if (st != fsymtab) {
+          // Same function but in a base class
+          overload_base = other;
+        }
+      }
+    }
+
+    // Check for shadowing/hiding base class function. Allow non-generic
+    // functions to shadow base class functions.
+    if (Node *other = Getattr(n, "hides")) {
+      if (Getattr(other, "fortran:generic")) {
+        Swig_warning(WARN_LANG_OVERLOAD_SHADOW, input_file, line_number,
+                     "Ignoring '%s' because it shadows a generic base class function '%s'\n",
+                     fsymname, Getattr(other, "fortran:name"));
+        Swig_warning(WARN_LANG_OVERLOAD_SHADOW, Getfile(other), Getline(other),
```

Comment:
WARN_LANG_OVERLOAD_SHADOW is implemented in the core and is being used in a different context here. It should be a different warning. I'm not entirely sure what it means too.

Reply:
Fixed in 1b405a480